source/common/http/filter/lua/lua_filter.h

```diff
@@ -279,6 +292,12 @@ class Filter : public StreamFilter, Logger::Loggable<Logger::Id::lua> {
     void continueIteration() override { return callbacks_->continueDecoding(); }
     void onHeadersModified() override { callbacks_->clearRouteCache(); }
     void respond(HeaderMapPtr&& headers, Buffer::Instance* body, lua_State* state) override;
+    const envoy::api::v2::core::Metadata routeEntryMetadata() const override {
+      if (callbacks_ && callbacks_->route() && callbacks_->route()->routeEntry()) {
+        return callbacks_->route()->routeEntry()->metadata();
```

Comment:
When using the metadata on from a  route, the filter name needs to be specified so it should be looking for 'envoy.lua' (using  Envoy::Config::HttpFilterNames::get().LUA)

Here is the metadata doc : https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route.proto.html?highlight=metadata
And an example from the router:
https://github.com/envoyproxy/envoy/blob/96c645cf5119934aadfdfad4d50622a081fc6ac3/source/common/router/config_impl.cc#L413-L414

Reply:
@ccaraman Thanks for your review pointing this out. 

Since this `callbacks_->route()`  returns [the route for the current request](https://github.com/envoyproxy/envoy/blob/master/include/envoy/http/filter.h#L106), could we just reuse it to get the route entry (then the corresponding metadata)?

Reply:
Apologies, I should have been more clear in my original comment. You are correct using the callback to get the metadata is the way to go. The missing step is filtering the metadata for entries associated with the Lua filter name.

Reply:
Ah OK! Sorry for didn't catch that. I've re-read the doc and made some changes. Thanks!